jan ide release note (#2804)
fishtown-analytics/docs.getdbt.com: website/docs/docs/dbt-versions/release-notes/01-Jan-2023/ide-updates.md

---
title: "Cloud IDE updates and fixes"
id: "ide-updates"
description: "Cloud IDE enhancements"

> sidebar_label: "IDE updates and fixes"

Reviewer: I'm not married to this wording, but I wonder if we should keep consistent with previous naming conventions? I update [this section](https://www.notion.so/dbtlabs/Request-a-dbt-Cloud-release-note-f962f39b49af4a23811d70984fef4f65#bf538d5b1cb44ba5ba58fb0411d80783) of our internal release notes page to capture these differences.  
```suggestion
sidebar_label: "Update and Fixes: IDE"
```

Author: oh good callout, thank you Leona! i should've noticed that 🙏

Learn more about the [January changes](https://getdbt.slack.com/archives/C03SAHKKG2Z/p1675272600286119) and what's coming soon.

## New features 

- Improved syntax highlighting within the IDE for better Jinja-SQL combination (double quotes now show proper syntax highlight!)
- Adjusted the routing URL for the IDE page and removed the `next` from the URL
- Added a *new* easter egg within the IDE 🐶🦆

## Product refinements 

- Performance improvements and reduced IDE slowness. The IDE should feel faster and snappier.
- Reliability improvements – Improved error handling that previously put IDE in a bad state
- Corrected the list of dropdown options for the Build button
- Adjusted startup page duration
- Added code snippets for `unique` and `not_null` tests for yml files
- Added code snippets for metrics based on environment dbt versions
- Changed “commit and push” to “commit and sync” to better reflect the action
- Improved error message when saving or renaming files to duplicate names

> ## Bugs fixed 

Reviewer: Just changing this to parallel other sections
```suggestion
## Bug fixes
```

> - Users running into `RPC server got an unknown async ID` arbitrarily

Reviewer: ```suggestion
- You no longer arbitrarily encounter an `RPC server got an unknown async ID` message
```

> - The build button dropdown was hidden behind the placeholder DAG screen

Reviewer: ```suggestion
- You can now see the build button dropdown, which had been hidden behind the placeholder DAG screen
```

> - Unable to close toast notifications for command failure when the history drawer is open

Reviewer: ```suggestion
- You can now close toast notifications for command failure when the history drawer is open
```

> - Users ran into `Something went wrong` when previewing a model

Reviewer: ```suggestion
- You no longer encounter a `Something went wrong` message when previewing a model
```

> - Unable to retrieve repository status in the IDE due to inability to find SSH folder

Reviewer: Was there an error message this addressed?
```suggestion
- You can now see repository status in the IDE, and the IDE finds the SSH folder
```

> - Scroll bars and download CSV flicker within the preview pane

Reviewer: ```suggestion
- Scroll bars and download CSV no longer flicker within the preview pane. 
```

> - Scroll bars and download CSV no longer flicker within the preview pane. 

Reviewer: ```suggestion
- Scroll bars and download CSV no longer flicker within the preview pane
```

## Coming soon 

- dbt Labs will roll out a series of features that should improve the quality of life within the IDE over the next several months, such as autosave, the ability to revert individual files, and user experience improvements, like right-clicking.

> - dbt Labs is researching ways to include Linting/SQL Fluff in the IDE. If anyone is interested in sharing how you're using SQL Fluff today, please reach out to dbt Labs Product Manager, [Nate Nunta](bossman's email or dm?).

Author: hey @bossnunta - if users want to reach out to you, would it only be via slack dm's or email or another option? thinking of users that may not have slack

```suggestion
- dbt Labs is researching ways to include Linting/SQL Fluff in the IDE. If anyone is interested in sharing how you're using SQL Fluff today, please reach out to dbt Labs Product Manager, [Nate Nunta](bossman's email or dm?).
```

Reviewer 2: I think let's leave my name in there but -- DM me on community Slack or use this email cloud-ide-feedback@dbtlabs.com
